src/main/java/io/github/filelize/path/PathHandler.java

```diff
@@ -75,10 +76,8 @@ private String getFilename(String id, String name) {
         return name + "_" + id + ".json";
     }
     private String getDirectoryPath(Object object){
-        if(object != null) {
-            return basePath + "/" + getFilelizeDirectory(object);
-        }
-        return basePath;
+        Objects.requireNonNull(object);
+        return basePath + "/" + getFilelizeDirectory(object);
```

Comment:
We can certainly trim that out. I'm definitely not able to invest in a deep dive, so a lot of changes are cursory.

Reply:
This is invalid. I added a test for `PathHandler` on a lark: 

```java
@Test
public void testPathHandlerReturnsEmptyBaseForNull() {
        var path=pathHandler.getFullPath2("foo", null);
        assertEquals("src/test/resources/foo.json", path);
}```

And BINGO! The exception from that check was thrown. It might be that `getFullPath2` is not intended for use, in which case it shouldn't be available or should be marked clearly. That exception is wrong, of course - it's a `NullPointerException` and should reflect the actual error, which is that persistence of a `null` is being requested - but the check's correct in semnatics.

Reply:
I updated the PR with code that *passes* and expects an NPE for this case. This is a specification thing: I don't know what you want the library to do if the object is `null`.

Reply:
approved, gj (y)